src/System.Management.Automation/engine/Modules/ModuleIntrinsics.cs

```diff
@@ -1291,38 +1248,40 @@ public static string GetModulePath(string currentProcessModulePath, string hklmM
                 {
                     currentProcessModulePath += hklmMachineModulePath; // += EVT.Machine
                 }
+
+#if !UNIX
+                // Add Windows Modules path
+                currentProcessModulePath += Path.PathSeparator;
+                currentProcessModulePath += GetWindowsPowerShellPSHomeModulePath();
+#endif
             }
             // EVT.Process exists
             // Now handle the case where the environment variable is already set.
             else
             {
-                // When SxS PS instance A starts SxS PS instance B, A's PSHome module path might be inherited by B. We need to remove that path from B
-                currentProcessModulePath = RemoveSxSPsHomeModulePath(currentProcessModulePath, personalModulePath, sharedModulePath, psHomeModulePath);
-
                 string personalModulePathToUse = string.IsNullOrEmpty(hkcuUserModulePath) ? personalModulePath : hkcuUserModulePath;
                 string systemModulePathToUse = string.IsNullOrEmpty(hklmMachineModulePath) ? psHomeModulePath : hklmMachineModulePath;
 
                 currentProcessModulePath = AddToPath(currentProcessModulePath, personalModulePathToUse, 0);
 
                 int insertIndex = -1;
-#if !UNIX
-                string windowsPowerShellModulePath = GetWindowsPowerShellPSHomeModulePath();
-                // If the Windows PowerShell Module path is already present, insert the system module path
-                // ($PSHOME/Modules) before it.
-                insertIndex = PathContainsSubstring(currentProcessModulePath, windowsPowerShellModulePath);
-#endif
-                currentProcessModulePath = AddToPath(currentProcessModulePath, systemModulePathToUse, insertIndex);
-            }
 
-            // if we reached this point - always add <Program Files> location to EVT.Process
-            // everything below is the same behaviour as WMF 4 code
+                insertIndex = PathContainsSubstring(currentProcessModulePath, personalModulePath);
+                if (insertIndex >= 0)
+                {
+                    insertIndex += personalModulePath.Length + 1;
+                }
 
-            // index of $PSHome\Modules in currentProcessModulePath
-            int indexOfPSHomeModulePath = PathContainsSubstring(currentProcessModulePath, psHomeModulePath);
+                currentProcessModulePath = AddToPath(currentProcessModulePath, sharedModulePath, insertIndex);
```

Comment:
Given the `if (insertIndex >= 0)` check, this line should be put in the if block.

Reply:
Looking at it now, all the `if` statements aren't needed since the preceding lines insert it.  In fact, we don't even need to search for the index since we know the length of the strings.  Will rewrite this.

Reply:
Actually, this isn't correct, need to check because the inherited path may have it but in different locations.

Reply:
So that line is outside of the `if` because if that path is not found, we can't insert the next one behind it and instead append it to the end.  `AddToPath` will append to end if insertIndex is -1.

Reply:
If you meant to search for `personalModulePathToUse` instead of `personalModulePath`, then it's guaranteed to be present, meaning `insertIndex` is always `>= 0`

Reply:
Got it.  Will remove the `if`